crates/sui-rosetta/src/construction.rs

```diff
@@ -306,13 +335,18 @@ pub async fn metadata(
                 .internal_operation
                 .try_into_data(ConstructionMetadata {
                     sender,
-                    coins: vec![],
+                    coins: gas_coin
```

Comment:
Why is this needed here and before it wasn't for the dry run?

Reply:
This has been added in an attempt to see if its addition would solve the `VMVerificationOrDeserializationError` error, which apparently it didn't. So probably in the final PR it will be removed.

Reply:
yeah that shouldn't be the case since it works for the other scenarios (and it's only for gas logic)